commands/restart.go

```diff
@@ -1,13 +1,53 @@
 package commands
 
 import (
+	"os"
+	"strings"
+
 	"github.com/docker/machine/log"
 
 	"github.com/codegangsta/cli"
 )
 
 func cmdRestart(c *cli.Context) {
+	var activeUrl string
+	if activeInList(c.Args()) {
+		activeUrl = os.Getenv("DOCKER_HOST")
+	}
+
 	if err := runActionWithContext("restart", c); err != nil {
 		log.Fatal(err)
 	}
+
+	if activeInList(c.Args()) && activeUrl != getMachineUrl(c) {
+		// TODO: hardcoded port here is a giant kludge...
+		var swarmOpt string
+		if strings.Contains(activeUrl, ":3736") {
```

Comment:
This should be `3376`.  More and more reason we need to remove the hardcode. :(  I think that just might be a goal of 0.4 :)